net.go

```diff
@@ -20,9 +20,17 @@ const minPortRange = 1
 
 const maxPortRange = 65535
 
+// IPFilterFunc allows to filter IP addresses.
+//
+// List of IPFilterFuncs is applied with AND logic: all filters should return true
+// for the IP address to be included in the response.
+type IPFilterFunc func(ip net.IP) bool
+
 // IPAddrs finds and returns a list of non-loopback IP addresses of the
 // current machine.
-func IPAddrs() (ips []net.IP, err error) {
+//
+// List of filters is optional, if any filter returns false, IP is skipped.
+func IPAddrs(filters ...IPFilterFunc) (ips []net.IP, err error) {
```

Comment:
I wonder if it might be better to keep `IPAddrs()` as it is and instead provide a more generic `IPFilter(list []net.IP, filters ...IPFilterFunc) (ips []net.IP, err error)` wrapper.

Reply:
Or: `func IPFilter(filters ...IPFilterFunc) func(ips []net.IP) ([]net.IP, error)`

Reply:
yeah, good point, I was torn on that. we already have `FilterIPs` with CIDRs, but SideroLink ULA can't be filtered with CIDR, so we need a function.

Reply:
So... `FilterIPs` and `FilterSubnets`?

Reply:
`IPFilter` sounds good to me, as we're filtering IPs here, more generic way would be to filter IPNets, but looks like we don't need it right now. 

Overall I'm waiting for Go 1.18 and a moment to rewrite all of that using stdlib's `netip`

Reply:
refactored with separate `IPFilter`